overlord/configstate/configcore/picfg.go

```diff
@@ -92,13 +95,53 @@ func updatePiConfig(path string, config map[string]string) error {
 	return nil
 }
 
-var (
-	errPiConfigNotSupported = fmt.Errorf("configuring pi-config not supported in current mode")
-)
+type piConfigNotSupportedError struct {
+	reason string
+}
+
+func newPiConfigNotSupportedError(msg string) *piConfigNotSupportedError {
+	return &piConfigNotSupportedError{msg}
+}
+
+func (e *piConfigNotSupportedError) Error() string {
+	return fmt.Sprintf("configuring not supported: %s", e.reason)
+}
+
+var reIgnorePrefix = regexp.MustCompile(`(?i)^#\s+Snapd-Edit:\s+no\s*$`)
+
+func piConfigFileIgnoreMarkerSet(configFile string) bool {
+	f, err := os.Open(configFile)
+	if err != nil {
+		return false
```

Comment:
perhaps it'd be useful to have a test which removes the file or makes it unreadable to make sure we don't do something silly

Reply:
Nice catch, it seems we are just silently doing nothing if there is no config.txt, no error and no new one is created. I think that is probably outside the scope of this PR as it seems we have this behavior since a long time. I guess it's also a bit of a question what the right behavior is in this case, probably error(?).

Reply:
remember that this code is called everywhere even on things that are not pis at all ATM, that's why I think it's like that

Reply:
I was confused, this is new code, in theory we could return an error if the error is not file does not exist, not sure it matters a lot in practice, as we will try to use the file again later and file again I suppose

Reply:
Yeah, I think so too, we will later read the file and fail with a error.